contrib/upgrade.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+set -e
+
+# This is an update script for gitea deployed from the binary distribution
+# from dl.gitea.io on linux as systemd service. It performs backup and updates
+# Gitea in place.
+# Depends on: bash, curl, xz, sha256sum, gpg, which. optionally jq.
+# Usage:      [environment vars] upgrade.sh [version]
+#   See section below for available environment vars.
+#   When no version is specified, updates to the latest release.
+# Examples:
+#   upgrade.sh 1.15.10
+#   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
+
+# apply variables from environment
+: ${giteabin:=/usr/local/bin/gitea}
+: ${giteahome:=/var/lib/gitea}
+: ${giteaconf:=/etc/gitea/app.ini}
+: ${giteauser:=git}
+: ${sudocmd:=sudo}
+: ${arch:=linux-amd64}
+
+function giteacmd {
+  "$sudocmd" -u "$giteauser" "$giteabin" -c "$giteaconf" -w "$giteahome" $@
+}
+
+function require {
+  for exe in $@; do
+    which $exe &>/dev/null || (echo "missing dependency '$exe'"; exit 1)
+  done
+}
+require curl xz sha256sum gpg
+
+# select version to install
+if [[ -z "$1" ]]; then
+  require jq
+	giteaversion=`curl -sL https://dl.gitea.io/gitea/version.json | jq -r .latest.version`
+else
+	giteaversion="${1}"
+fi
+
+# confirm update
+current=`giteacmd --version | cut -d' ' -f3`
```

Comment:
```suggestion
current=$(giteacmd --version | cut -d' ' -f3)
```
Let's not mix styles. I think we should avoid ` as it's legacy.

Reply:
Also, I just noticed that it would be beneficial to add a check here whether `"$current" == "$giteaversion"` to prevent misuse/ no-ops.
In this specific case, it doesn't even matter whether `"$current" > "$giteaversion"`, because this script in the current implementation can be also used to downgrade gitea.
(Not even an additional warning is needed as the user will already be asked for confirmation below anyway)